src/main/java/ch/njol/skript/expressions/ExprPotionEffect.java

```diff
@@ -20,36 +20,79 @@
 
 import org.bukkit.event.Event;
 import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
 import org.eclipse.jdt.annotation.Nullable;
 
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.Timespan;
 import ch.njol.util.Kleenean;
 
-/**
- * @author Peter Güttinger
- */
+@Name("Potion Effect")
+@Description({"Create a new potion effect to apply to an entity or item type. Do note that when applying potion effects ",
+	"to tipped arrows/lingering potions, Minecraft reduces the timespan."})
+@Examples({"set {_p} to potion effect of speed of tier 1 without particles for 10 minutes",
+	"add {_p} to potion effects of player's tool",
+	"add {_p} to potion effects of target entity",
+	"add potion effect of speed 1 to potion effects of player"})
+@Since("INSERT VERSION")
 public class ExprPotionEffect extends SimpleExpression<PotionEffect> {
```

Comment:
Maybe EffPotion should also (or instead) take in PotionEffects?
With this, you kind of have two systems that IMO need to be brought together somehow.

Reply:
Ok, I added support for potion effects in EffPotion. I decided to prevent breakages and just add it rather than replacing what is currently there.